### PR TITLE
docs: add pnpm tip to LCP optimisation 💡

### LINKED ISSUE #29272: LCP Advanced optimizations hint for pnpm
### Describe the feature

This is just a small adaption, but it might be nice to add a small hint for pnpm users in the docs for LCP improvements.

📘 [docs link](https://nuxt.com/docs/getting-started/styling#lcp-advanced-optimizations)

### Additional information

- [x] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [x] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [x] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by dreitzner**
### 🔗 Linked issue

resolves #29272 

### 📚 Description

The current docs for LCP optimizations do not work with pnpm.

This adds a hint to the docs for all pnpm users.

[📘 Current docs](https://nuxt.com/docs/getting-started/styling#lcp-advanced-optimizations)

### CODE REVIEW

File: docs/1.getting-started/4.styling.md [hunk 1]
```diff
@@ -542,3 +542,8 @@ export default defineNuxtConfig({
   },
 })
 ```
+
+::tip
+If you are using pnpm, the actual path will be different.
+Read this [blog post](https://dev.to/dreitzner/nuxt3-lcp-advanced-optimizations-with-pnpm-12h3) to get started.
```
Comment (HigherOrderLogic) [inline docs/1.getting-started/4.styling.md:548]: I think the blog is quite short and is possible to fit into this docs page as a separate section, without the need of linking to other site.

Comment (danielroe) [reply docs/1.getting-started/4.styling.md:548]: let's update the code above instead, so it works even on pnpm 🙏

Comment (dreitzner) [reply docs/1.getting-started/4.styling.md]: @danielroe thank you for the feedback. 

I pushed the change with the code directly included now.

File: docs/1.getting-started/4.styling.md [hunk 2]
```diff
@@ -542,3 +542,34 @@ export default defineNuxtConfig({
...
+        // start from the end of the array and go to the beginning
+        for (let i = css.length - 1; i >= 0; i--) {
+          // if it starts with 'entry', remove it from the list
+          if (css[i].startsWith('entry')) css.splice(i, 1);
+        }
+      }
+    },
+  },
+})
+```
```
Comment (HigherOrderLogic) [inline docs/1.getting-started/4.styling.md:575]:
````suggestion
```ts [nuxt.config.ts]
import { readdirSync } from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';

export default defineNuxtConfig({
  hooks: {
    'build:manifest': (manifest) => {
      // find the app entry, css list
      const baseModulesFolder = path.dirname(fileURLToPath(import.meta.url)) + '/node_modules/.pnpm/';
      const nuxtFolder = readdirSync(baseModulesFolder).find((file) => typeof file === 'string' && file.startsWith('nuxt@'));
      const entryFilePath = `node_modules/.pnpm/${nuxtFolder}/node_modules/nuxt/dist/app/entry.js`;

      const css = manifest[entryFilePath]?.css;
      if (css) {
        // start from the end of the array and go to the beginning
        for (let i = css.length - 1; i >= 0; i--) {
          // if it starts with 'entry', remove it from the list
          if (css[i].startsWith('entry')) css.splice(i, 1);
        }
      }
    },
  },
})
```
::
````

Is it possible to render the whole code block inside the tip section?

File: docs/1.getting-started/4.styling.md (hunk 1 above)
Comment (danielroe) [reply docs/1.getting-started/4.styling.md]: I'm suggesting instead that we don't have a workaround for pnpm. Instead, we update the code block above it so it works equally well on pnpm and non-pnpm projects

You can access `chunk.isEntry` IIRC

Comment (dreitzner) [reply docs/1.getting-started/4.styling.md]: @danielroe thank you for pointing that out.

I psuhed the changes

File: docs/1.getting-started/4.styling.md [hunk 3]
```diff
@@ -530,7 +530,11 @@ export default defineNuxtConfig({
   hooks: {
     'build:manifest': (manifest) => {
       // find the app entry, css list
-      const css = manifest['node_modules/nuxt/dist/app/entry.js']?.css
+      const [
+        entryFilePath = '',
+        _entryFileOptions
+      ] = Object.entries(manifest).find(([_path, options]) => options.isEntry) ?? [];
+      const css = manifest[entryFilePath]?.css;
```
Comment (danielroe) [inline docs/1.getting-started/4.styling.md:537]:
```suggestion
      const css = Object.values(manifest).find(options => options.isEntry)?.css
```

**Review by danielroe — APPROVED** [review]

**Pull request merged**